pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -306,10 +306,14 @@ func (s *DiscoveryServer) periodicRefreshMetrics() {
 	defer ticker.Stop()
 	for range ticker.C {
 		push := s.globalPushContext()
+		push.Mutex.Lock()
 		if push.End != timeZero {
+			model.LastPushMutex.Lock()
 			model.LastPushStatus = push
```

Comment:
Should also protect the `LastPushStatus ` read?

Reply:
It's a write/write data race

Reply:
It's a write/write data race:

```sh
WARNING: DATA RACE
Write at 0x000004918d60 by goroutine 33:
  istio.io/istio/pilot/pkg/proxy/envoy/v2.(*DiscoveryServer).periodicRefreshMetrics()
      /Users/ymesika/go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/discovery.go:304 +0x194

Previous write at 0x000004918d60 by goroutine 105:
  istio.io/istio/pilot/pkg/model.(*PushContext).OnConfigChange()
      /Users/ymesika/go/src/istio.io/istio/pilot/pkg/model/push_context.go:289 +0x4e
  istio.io/istio/pilot/pkg/proxy/envoy/v2.(*DiscoveryServer).Push()
      /Users/ymesika/go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/discovery.go:323 +0x990
  istio.io/istio/pilot/pkg/proxy/envoy/v2.(*DiscoveryServer).doPush()
      /Users/ymesika/go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/discovery.go:434 +0xf6
  istio.io/istio/pilot/pkg/proxy/envoy/v2.(*DiscoveryServer).doPush-fm()
      /Users/ymesika/go/src/istio.io/istio/pilot/pkg/proxy/envoy/v2/discovery.go:452 +0x4a
```

Reply:
Didn't encounter another race other than that